src/lib.rs

```diff
@@ -138,26 +223,30 @@ where
   zn_secondary: G2::Scalar,
   _p_c1: PhantomData<C1>,
   _p_c2: PhantomData<C2>,
+  a1: PhantomData<A1>,
```

Comment:
Should these be `_p_a1` and `_p_a2`?

Reply:
These fields are actually used to simplify initializing `IO::Vals` later.